kani-driver/src/call_goto_synthesizer.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::util::warning;
+use anyhow::Result;
+use std::ffi::OsString;
+use std::path::Path;
+use std::process::Command;
+
+use crate::session::KaniSession;
+
+impl KaniSession {
+    /// Synthesize loop contracts for a goto binary `input` and produce a new goto binary `output`
+    /// The synthesizer we use is `goto-synthesizer` built in CBMC codebase, which is an enumerative
+    /// loop-contracts synthesizer. `goto-synthesizer` enumerates and checks if a candidate can be
+    /// used to prove some assertions, and applies found invariants when all checks pass.
+    pub fn synthesize_loop_contracts(&self, input: &Path, output: &Path) -> Result<()> {
+        if !self.args.quiet {
+            println!("Running loop contract synthesizer.");
+            warning("WARNING: this process may not terminate.");
+        }
+
+        let args: Vec<OsString> = vec![
+            "--loop-contracts-no-unwind".into(),
+            input.to_owned().into_os_string(),  // input
+            output.to_owned().into_os_string(), // output
+        ];
+
+        let mut cmd = Command::new("goto-synthesizer");
+        cmd.args(args);
+
+        if self.args.quiet {
+            self.run_suppress(cmd)?;
+        } else {
+            self.run_terminal(cmd)?;
+        }
```

Comment:
You don't need this. In fact, `run_suppress` invokes `run_terminal` if `quiet` is true.

Reply:
My bad (this was my suggestion) :(

Reply:
We should clean up this instance then: https://github.com/model-checking/kani/blob/035225ca362b0a80c807af0f567c3513237cb0ee/kani-driver/src/concrete_playback.rs#L191